### PR TITLE
Handling data url's in Spritsheets

### LINKED ISSUE #2437: spritesheetParse is not accounting for base64 images
I'm trying to load a spritesheet that uses a base64 image. However, Pixi is assuming that it's going to be a file path so it's trying to append a baseUrl.

https://github.com/pixijs/pixi.js/blob/master/src/loaders/spritesheetParser.js#L29

### PR DESCRIPTION
**Pull request by davearel**
Fixes #2437

### CODE REVIEW

File: src/loaders/spritesheetParser.js [hunk 1]
```diff
@@ -23,10 +24,17 @@ module.exports = function ()
             metadata: resource.metadata.imageMetadata
         };
 
+        // Prepend url path unless the resource image is a data url
+        if (Resource.isDataUrl(resource.url)) {
```
Comment (englercj) [inline src/loaders/spritesheetParser.js]: Did you try this? `Resource.isDataUrl` is a boolean property, not a function.

https://github.com/englercj/resource-loader/blob/ba23abc5fd4662136aacee40cab2e565a611086a/src/Resource.js#L52

File: src/loaders/spritesheetParser.js [hunk 2]
```diff
@@ -23,10 +24,17 @@ module.exports = function ()
...
+            resourcePath = resource.data.meta.image;
+        } else {
+            resourcePath = path.dirname(resource.url.replace(this.baseUrl, '')) + '/' + resource.data.meta.image;
+        }
+
         var route = path.dirname(resource.url.replace(this.baseUrl, ''));
```
Comment (englercj) [inline src/loaders/spritesheetParser.js]: Doesn't look like `route` is used anymore, should remove it.

File: src/loaders/spritesheetParser.js [hunk 3]
```diff
@@ -23,10 +24,17 @@ module.exports = function ()
             metadata: resource.metadata.imageMetadata
         };
 
+        // Prepend url path unless the resource image is a data url
+        if (Resource.isDataUrl(resource.url)) {
+            resourcePath = resource.data.meta.image;
+        } else {
```
Comment (englercj) [inline src/loaders/spritesheetParser.js]: Please match the brace style used throughout PIXI:

```js
if ()
{
}
else
{
}
```

File: src/loaders/spritesheetParser.js (hunk 1 above)
Comment (davearel) [inline src/loaders/spritesheetParser.js]: Sorry, I noticed `isDataUrl` last minute. But evidently I was a little presumptuous. I'll replace with:
```javascript
if (resource.isDataUrl)
```

**Comment by davearel** [comment]
@englercj all set

**Comment by ivanpopelyshev** [comment]
Nice :) Im poking @GoodBoyDigital

**Comment by davearel** [comment]
Thanks guys!

**Pull request merged**